p8e-api/src/main/kotlin/io/provenance/engine/event/IndexHandler.kt

```diff
@@ -114,6 +123,33 @@ class IndexHandler(
                                     envelope.data.result.contract.recitalsList.map { it.signer.encryptionPublicKey.toPublicKey().toSha512Hex() }
                                 ), RequestOptions.DEFAULT
                             )
+                            // If the env is the invoker, create the data access message and put into a job.
+                            if(envelope.isInvoker == true && envelope.data.input.affiliateSharesList.isNotEmpty()) {
+                                val envelopeDataAccess = envelope.data.input.affiliateSharesList.map {
```

Comment:
This function is already kind of massive, I think it might be nice if this data access block got moved into a separate (private) function with some lovely name in this class (whether just the bit inside of the if, or a function that contains the if statement as well, and otherwise is a no-op), just to make it a bit easier to parse what is happening when reading